benchmarks/kotlin-ktor/src/main/kotlin/org/renaissance/kotlin/ktor/KtorRenaissanceBenchmark.kt

```diff
@@ -0,0 +1,120 @@
+
+import io.ktor.server.engine.*
+import kotlinx.coroutines.*
+import org.renaissance.Benchmark
+import org.renaissance.Benchmark.*
+import org.renaissance.BenchmarkContext
+import org.renaissance.BenchmarkResult
+import org.renaissance.BenchmarkResult.Validators
+import org.renaissance.License
+import org.renaissance.kotlin.ktor.client.ClientManager
+import org.renaissance.kotlin.ktor.server.ChatApplication
+import kotlin.math.min
+
+@Group("web")
+@Name("kotlin-ktor")
+@Summary("Simple Ktor chat application with multiple clients, performing various tasks.")
+@Licenses(License.MIT)
+@Parameter(
+  name = "port",
+  defaultValue = "8080",
+  summary = "Port to run the server on."
+)
+@Parameter(
+  name = "client_count",
+  defaultValue = "\$cpu.count",
+  summary = "Number of clients that are simultaneously sending the requests"
+)
+@Parameter(
+  name = "iterations_count",
+  defaultValue = "2000",
+  summary = "Number of times clients should repeat their designated operations"
+)
+@Parameter(
+  name = "chat_count",
+  defaultValue = "10",
+  summary = "How many public chats should be setup for user interactions." +
+      "Reducing/increasing number of chats will increase/decrease runtime"
+)
+@Parameter(
+  name = "group_message_fraction",
+  defaultValue = "1.0",
+  summary = "Clients which execute joinChatAndSendMessage"
+)
+@Parameter(
+  name = "private_message_fraction",
+  defaultValue = "0.5",
+  summary = "How many public chats should be setup for user interactions"
+)
+@Parameter(
+  name = "random_seed",
+  defaultValue = "32",
+  summary = "Random seed to use for client tasks setup."
+)
+class KtorRenaissanceBenchmark() : Benchmark {
+  private var port: Int = 0
+  private var clientCount: Int = 0
+  private var numberOfRepetitions: Int = 0
+  private var fractionOfClientsSendingPrivateMessages: Double = 0.0
+  private var fractionOfClientsSendingGroupMessages: Double = 0.0
+  private lateinit var clientPool: ExecutorCoroutineDispatcher
+  private lateinit var server: ApplicationEngine
+  private lateinit var application: ChatApplication
+  private lateinit var clientManager: ClientManager
+
+  @OptIn(DelicateCoroutinesApi::class)
+  override fun setUpBeforeAll(context: BenchmarkContext?) {
+    port = context!!.parameter("port").toPositiveInteger()
+    clientCount = context.parameter("client_count").toPositiveInteger()
+    numberOfRepetitions = context.parameter("iterations_count").toPositiveInteger()
+    val numberOfChats = context.parameter("chat_count").toPositiveInteger()
+    fractionOfClientsSendingGroupMessages = context.parameter("group_message_fraction").toDouble()
+    fractionOfClientsSendingPrivateMessages =
+      context.parameter("private_message_fraction").toDouble()
+    val randomSeed = context.parameter("random_seed").toPositiveInteger()
+
+    application = ChatApplication(numberOfChats)
+    server = embeddedServer(io.ktor.server.cio.CIO, host = "127.0.0.1", port = port) {
```

Comment:
You should use `localhost` instead of `127.0.0.1`, otherwise it may fail on IPv6-only environments.